pkg/apiserver/apiserver.go

```diff
@@ -85,7 +86,10 @@ func NewAPIServer() *cobra.Command {
 	cmd.Flags().StringVar(&flags.TLSCertFile, "tls-cert-file", "", "File containing the default x509 Certificate for HTTPS. If not provided no TLS security shall be enabled")
 	cmd.Flags().StringVar(&flags.TLSPrivateKeyFile, "tls-private-key-file", "", "File containing the default x509 private key matching --tls-cert-file.")
 	cmd.Flags().StringArrayVar(&flags.CORSAllowedOrigins, "cors-allowed-origins", []string{"*"}, "List of allowed origins for CORS, comma separated. An allowed origin can be a regular expression to support subdomain matching. If this list is empty CORS will not be enabled.")
-	oidc.AddOIDCPFlags(&flags.OIDCOptions, cmd.Flags())
+	cmd.Flags().BoolVar(&flags.enableOIDC, "enable-oidc", false, "List of allowed origins for CORS, comma separated. An allowed origin can be a regular expression to support subdomain matching. If this list is empty CORS will not be enabled.")
```

Comment:
What about we define this as a list of auth-providers?

e.g. `--authentication-mode=OIDC,Static,ServiceAccount`

ref from Kubernetes:
```
--authorization-mode stringSlice     Default: [AlwaysAllow]
--
  | Ordered list of plug-ins to do authorization on secure port. Comma-delimited list of: AlwaysAllow,AlwaysDeny,ABAC,Webhook,RBAC,Node.
```

Reply:
Yeah, I put it as a simple boolean because Neven has already done that in #459, and I don't want to repeat that PR here, after that PR is merged, will also adapt the changes here.

Reply:
done